cpp/src/arrow/ipc/writer.h

```diff
@@ -298,11 +299,13 @@ namespace internal {
 
 // These internal APIs may change without warning or deprecation
 
-// Intermediate data structure with metadata header plus zero or more buffers
-// for the message body. This data can either be written out directly as an
+// Intermediate data structure with optional FlightDescriptor,
```

Comment:
I find it weird that we add a FlightDescriptor buffer in our Flight-agnostic IPC format. @wesm, what's your take on this?

Reply:
Yes, I also find it weird. We should use composition with `IpcPayload` in the Flight implementation instead if possible

Reply:
I'm going to take a quick look and see if we can avoid this modification